File: scripts/dmzMBRAPluginNAEdit.lua
local NodeType = dmz.object_type.new ("na_node")

function receive (self, type, data)
   if dmz.data.is_a (data) then
      local handle = data:lookup_handle ("object", 1)
      if dmz.object.is_object (handle) then 
         if dmz.object.type (handle):is_of_type (NodeType) then
            self.message:send ("NetworkAnalysisNodeProperties", data)
         else
            self.message:send ("NetworkAnalysisLinkProperties", data)
         end
      elseif dmz.object.is_link (handle) then
         self.message:send ("NetworkAnalysisLinkProperties", data)
      end
   end
end

function new (config, name)

   local self = {
      log = dmz.log.new ("lua." .. name),
      message = config:to_message ("edit.name", "EditObjectAttributesMessage"),
      obs = dmz.message_observer.new (name),
   }

   self.log:info ("Creating plugin: " .. name)
   self.obs:register (self.message, receive, self)

   return self
end

File: scripts/dmzMBRAPluginNAEdit.js
var dmz =
      { object: require("dmz/components/object")
      , objectType: require("dmz/runtime/objectType")
      , defs: require("dmz/runtime/definitions")
      , data: require("dmz/runtime/data")
      , message: require("dmz/runtime/messaging")
      }
   , NodeType = dmz.objectType.lookup("na_node")
   , nanpHandle = dmz.defs.createNamedHandle("NetworkAnalysisNodeProperties")
   , nalpHandle = dmz.defs.createNamedHandle("NetworkAnalysisLinkProperties")
   //, NodeType = dmz.objectType.lookup("na_node")
   , MsgType = dmz.message.create(
      self.config.string("edit.name","EditObjectAttributesMessage"))
   ;

// Open Edit NA node window

// NOTE:
// Necessary to convert strings into new named handles before sending messages

MsgType.subscribe (self,function (data) {
   var handle
     , nanpHandle
     , nalpHandle
     ;
   if (dmz.data.isTypeOf (data)) {
      var handle = data.handle("object", 0);
      if (dmz.object.isObject (handle)) {
         if (dmz.object.type(handle).isOfType(NodeType)) {
            MsgType.send(nanpHandle, data);
         }
         else {
            MsgType.send(nalpHandle, data);
         }
      }
      else if (dmz.object.isLink (handle)) {
         MsgType.send(nalpHandle, data);
      }
   }
});

